metapool/tests/test_sample_sheet.py

```diff
@@ -1731,6 +1750,440 @@ def test_metagenomic_sheet_creation(self):
+                    'QiitaID': '101',
+                    'BarcodesAreRC': 'False',
+                    'ForwardAdapter': 'GATACA',
+                    'ReverseAdapter': 'CATCAT',
+                    'HumanFiltering': 'False',
+                    'library_construction_protocol': 'Knight Lab Kapa HP',
+                    'experiment_design_description': 'some description',
+                    'contains_replicates': 'False'
+                }
+            ],
+            'Contact': [
+                {
+                    'Sample_Project': 'MyProject_99999',
+                    'Email': 'foo@bar.org'
+                }
+            ],
+            'Assay': 'Metagenomic',
+            'SheetType': 'standard_metag',
+            'SheetVersion': '101'
+        }
+
+        data = [
+            ['sample1', 'sample1', 'A', 1, False, 'THDMI_10317_PUK2',
+             'MyProject_99999', 'THDMI_10317_UK2-US6', 'A1', '1', '1', 'SF',
+             '166032128', 'Carmen_HOWE_KF3', '109379Z', '2021-08-17', '978215',
+             'RNBJ0628', 'Echo550', 'THDMI_UK_Plate_2', 'THDMI UK', '', '1',
+             'A1', '515rcbc0', 'AATGATACGGCGACCACCGAGATCTACACGCT',
+             'AGCCTTCGTCGC', 'TATGGTAATT', 'GT', 'GTGYCAGCMGCCGCGGTAA',
+             'AATGATACGGCGACCACCGAGATCTACACGCTAGCCTTCGTCGCTATGGTAATTGTGTGYCAG'
+             'CMGCCGCGGTAA', 'pool1']
+        ]
+
+        columns = ['sample sheet Sample_ID',
+                   'Sample', 'Row', 'Col', 'Blank', 'Project Plate',
+                   'Project Name', 'Compressed Plate Name', 'Well',
+                   'Plate Position', 'Primer Plate #', 'Plating',
+                   'Extraction Kit Lot', 'Extraction Robot', 'TM1000 8 Tool',
+                   'Primer Date', 'MasterMix Lot', 'Water Lot',
+                   'Processing Robot', 'Sample Plate', 'Project_Name',
+                   'Original Name', 'Plate', 'EMP Primer Plate Well', 'Name',
+                   "Illumina 5' Adapter", 'Golay Barcode',
+                   'Forward Primer Pad', 'Forward Primer Linker',
+                   '515FB Forward Primer (Parada)', 'Primer For PCR',
+                   'syndna_pool_number']
+
+        table = pd.DataFrame(columns=columns, data=data)
+        sheet = make_sample_sheet(metadata, table, 'iSeq', [1], strict=False)
+
+        # confirm that we get a sample-sheet w/out katharoseq-control-related
+        # columns.
+        self.assertIsNotNone(sheet)
+        self.assertIsInstance(sheet, MetagenomicSampleSheetv101)
+        self.assertFalse(sheet.contains_katharoseq_samples())
+        obs_columns = set(sheet.samples[0].to_json().keys())
+        exp_columns = {'Sample_ID', 'Sample_Name', 'Sample_Plate',
+                       'well_id_384', 'I7_Index_ID', 'index', 'I5_Index_ID',
+                       'index2', 'Sample_Project', 'Well_description',
+                       'Lane'}
+        self.assertEqual(obs_columns, exp_columns)
+
+        ###
```

Comment:
rm? If you need different tests, please add another tests not this none obvious separations.

Reply:
It's more illustrative as a single test, IMHO. However I'll break them up.